pkg/client/transport/transport.go

```diff
@@ -55,9 +55,6 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	if !(c.HasCA() || c.HasCertAuth() || c.TLS.Insecure) {
 		return nil, nil
 	}
-	if c.HasCA() && c.TLS.Insecure {
```

Comment:
we don't want to change it here, we'd want to make sure the CA was cleared when handling the `--insecure-skip-tls-verify` check